mlir/lib/Dialect/Transform/IR/TransformInterfaces.cpp

```diff
@@ -1444,16 +1443,20 @@ bool transform::ErrorCheckingTrackingListener::failed() const {
 }
 
 void transform::ErrorCheckingTrackingListener::notifyPayloadReplacementNotFound(
-    Operation *op, ValueRange values) {
+    Operation *op, ValueRange values, ArrayRef<Operation *> aliveUsers) {
   if (status.succeeded()) {
     status = emitSilenceableFailure(
-        getTransformOp(), "tracking listener failed to find replacement op");
```

Comment:
maybe "tracking listener failed to find replacement op for op that is tracked by non-dead handle".

But even that could be confusion. Maybe we should just add a comment in the C++ code that mentions all the conditions (type changed, handle not dead, etc.).

Reply:
I was thinking literally returning `DiagnosedSilecenableFailure` instead of `LogicalResult`, then the replacement lookup can specify the message.